src/posix/main.c

```diff
@@ -67,9 +67,12 @@
 #include <openthread/cli.h>
 
 #include "cli/cli_config.h"
-#if (HAVE_LIBEDIT || HAVE_LIBREADLINE) && !OPENTHREAD_POSIX_CONFIG_DAEMON_ENABLE && \
-    (OPENTHREAD_CONFIG_CLI_TRANSPORT == OT_CLI_TRANSPORT_CONSOLE)
+#if OPENTHREAD_CONFIG_CLI_TRANSPORT == OT_CLI_TRANSPORT_CONSOLE
 #define OPENTHREAD_USE_CONSOLE 1
+#if !(HAVE_LIBEDIT || HAVE_LIBREADLINE) || OPENTHREAD_POSIX_CONFIG_DAEMON_ENABLE
```

Comment:
move this block into console_cli.h?

Reply:
SGTM. I also moved the above default definition of `HAVE_LIBEDIT` and `HAVE_LIBREADLINE` into `console_cli.h` to avoid warnings.